src/actor/component.rs

```diff
@@ -52,9 +55,18 @@ impl host::Host for Ctx<'_> {
         operation: String,
         payload: Option<Vec<u8>>,
     ) -> anyhow::Result<Result<Option<Vec<u8>>, String>> {
-        let invocation = (namespace, operation, payload)
+        let invocation: Invocation = (namespace, operation, payload)
```

Comment:
```suggestion
        let invocation = (namespace, operation, payload)
```

The bound will be redundant once you rework the code below

Reply:
Personally I think making generic types in the call context is overkill and forces the consumer to do more gymnastics to get stuff done. The appeal of this type of context is that it's opaque and doesn't push complexity where it doesn't belong.

I get that it looks "cleaner" if you love generics but having been down this road before it feels like it might make things harder down the road.